### ISSUE TITLE
Publish public-codebase from-scratch Benchmark alongside internal-branch results

### ISSUE THREAD

**Issue by levipereira** [ultralytics/ultralytics#25876]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

The [YOLO26 Training Recipe](https://docs.ultralytics.com/guides/yolo26-training-recipe) page states:

> Not exactly — the checkpoints were produced using an internal training branch with additional features not in the public codebase (like configurable o2m weights and cls_w). You can get very close results using the hyperparameters documented on this page with the public Ultralytics package, but an exact reproduction requires the internal branch.

This means the officially published mAP values (e.g. 48.6 mAP50-95 for YOLO26s on COCO) cannot currently be reproduced by anyone using only the public ultralytics package — not even with the exact hyperparameters listed on that page, since key training features (o2m weights, cls_w) simply aren't in the public codebase.

This has already surfaced independently in #23971, where a user training from scratch with the public package, using the documented recipe, plateaued far below the published benchmark even after 1300+ epochs.

Request: Please publish a second set of from-scratch benchmark numbers — trained using only the public codebase (no internal-branch features) — alongside the current official numbers. Right now the published metrics function as an internal-branch ceiling that's not achievable or verifiable by external users, which undermines the benchmark table's usefulness as a comparison/reproducibility reference. Having both numbers side by side (internal-branch result vs. public-repo-reproducible result) would let users validate their own training runs against a realistic, achievable baseline.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @levipereira, thank you for clearly outlining the reproducibility gap between the published YOLO26 benchmarks and results achievable with the public Ultralytics codebase. Publishing a separate public-repository benchmark alongside the internal-branch results would provide a more useful and verifiable baseline for the community.

We’ll pass this request to the team for review alongside the [YOLO26 Training Recipe](https://docs.ultralytics.com/guides/yolo26-training-recipe) and the related issue you referenced. This is an automated response, and an Ultralytics engineer will assist soon.

**Comment by Y-T-G** [comment]
Hello.

The user in the linked issue didn't follow the YOLO26 training recipe at all, so it's not a valid issue to point to.

The performance of YOLO26 model can be reproduced by the Ultralytics repo. The disclaimer only says it wouldn't be bit-identical. But it would be within negligible distance from the official values. In fact, it may even be better because there have been significant bug fixes in the training pipeline since the YOLO26 release.

**Comment by levipereira** [comment]
Fair points on both —the linked issue isn't the strongest evidence if that user's setup deviated from the recipe, and I take the point about bug fixes since release potentially closing (or even flipping) the gap.

That said, my main ask stands independently of whether performance is reproducible "within negligible distance." Across the ML community, it's standard practice for released benchmark checkpoints to ship with full training logs — loss curves and mAP progression epoch-by-epoch, not just the final aggregated hyperparameters embedded in the .pt file. Previous YOLO releases (v5, v8, and others in the ecosystem) followed this practice.

The results (48.6 mAP50-95 for YOLO26s, etc.) are already public. The high-level hyperparameters are already public on the Training Recipe page. Given that, I don't see a reason to withhold the actual training logs — this isn't a request to expose the internal branch's code or the unreleased features (o2m weights, cls_w), just the run-level data that lets the community independently verify convergence behavior and compare it against their own from-scratch runs using the public codebase.

**Comment by Y-T-G** [comment]
> That said, my main ask stands independently of whether performance is reproducible "within negligible distance." Across the ML community, it's standard practice for released benchmark checkpoints to ship with full training logs — loss curves and mAP progression epoch-by-epoch, not just the final aggregated hyperparameters embedded in the .pt file. Previous YOLO releases (v5, v8, and others in the ecosystem) followed this practice.

All of these are public. The curves for Ultralytics models are stored inside the model checkpoints.

And it's also visible on Ultralytics Platform, that parses it from the model, if you don't want to visualize it manually:
https://platform.ultralytics.com/ultralytics/yolo26

Even the "internal" branch is public and you can just checkout the commit hash to use it, also stored in the model checkpoint:

```python
from ultralytics import YOLO

model = YOLO("yolo26n.pt")
print(model.ckpt.get("git"))
```

`git checkout cb13d5f9cfbd6f299da3620c625f81d721dc2849`

**Comment by levipereira** [comment]
Thanks for the pointers — appreciate the transparency around the git hash and the Platform link.

I checked, though, and the logs embedded in the checkpoint only cover ~70 epochs, not the full from-scratch curve. YOLOv10 published complete epoch-by-epoch logs for their from-scratch runs ([THU-MIG/yolov10/logs](https://github.com/THU-MIG/yolov10/tree/main/logs)), which is the kind of granularity I was hoping for here.

That said, I won't push further on this — just wanted to flag the gap. Publishing the raw values directly would honestly be the simplest path if the team ever revisits it, but I understand if it's not a priority.

<img width="455" height="297" alt="Image" src="https://github.com/user-attachments/assets/4d7aeeb7-f792-48ba-93b4-30c214da0286" />

**Comment by levipereira** [comment]
I don't quite understand why publishing the from-scratch results would be difficult — the rest of the community does this routinely, unless the model itself becomes private/proprietary, which I'd understand. But short of that, withholding these numbers seems a bit odd.

If I train from scratch today, what value will I actually reach, epoch by epoch, up to the final epoch? That's still an unknown — and it's exactly the kind of question a published from-scratch curve would answer.

**Comment by Y-T-G** [comment]
> 
> Thanks for the pointers — appreciate the transparency around the git hash and the Platform link.
> 
> I checked, though, and the logs embedded in the checkpoint only cover ~70 epochs, not the full from-scratch curve. YOLOv10 published complete epoch-by-epoch logs for their from-scratch runs ([THU-MIG/yolov10/logs](https://github.com/THU-MIG/yolov10/tree/main/logs)), which is the kind of granularity I was hoping for here.
> 
> That said, I won't push further on this — just wanted to flag the gap. Publishing the raw values directly would honestly be the simplest path if the team ever revisits it, but I understand if it's not a priority.
> 
> <img width="455" height="297" alt="Image" src="https://github.com/user-attachments/assets/4d7aeeb7-f792-48ba-93b4-30c214da0286" />
> 

That's the correct curve. It was trained for 70 epochs. I am not sure why you think it's some partial curve.

YOLO26 wasn't trained from scratch. It was trained from pretrained checkpoints trained on Objects365 (which are also public and documented on the Object365 page: https://docs.ultralytics.com/datasets/detect/objects365). If you want details about training, you can read the YOLO26 paper:

https://arxiv.org/abs/2606.03748

**Comment by Y-T-G** [comment]
> I don't quite understand why publishing the from-scratch results would be difficult — the rest of the community does this routinely, unless the model itself becomes private/proprietary, which I'd understand. But short of that, withholding these numbers seems a bit odd.
> 
> If I train from scratch today, what value will I actually reach, epoch by epoch, up to the final epoch? That's still an unknown — and it's exactly the kind of question a published from-scratch curve would answer.

There's no from scratch logs, because YOLO26 models weren't trained from scratch on COCO. They were pretrained on Object365. This is explicitly mentioned in the published YOLO26 paper. You're working based on wrong assumptions and then incorrectly claiming something is being withheld.

Nothing is being withheld. Rather, you're asking for curves and logs that never existed. The curves you're seeing on Ultralytics Platform are exactly the ones that were produced by training.

**Comment by levipereira** [comment]
Thanks for clarifying — that lines up with what's documented in the paper  Objects365-v1 pretrain (150 epochs) feeding into the COCO fine-tune, plus an internal cls_w weight (3.48 for the s-scale) not exposed in the public config. Explains the gap.
